modules/caddyhttp/app.go

```diff
@@ -374,6 +385,17 @@ func (app *App) Start() error {
 	}
 
 	for srvName, srv := range app.Servers {
+		// TODO why is this needed for caddy respond now?
+		if srv.Socket == nil {
+			srv.Socket = make([]*string, len(srv.Listen))
+		}
+		if srv.Protocols == nil {
+			srv.Protocols = make([][]string, len(srv.Listen))
+			for pi, _ := range srv.Protocols {
+				srv.Protocols[pi] = []string{"h1", "h2", "h3"}
```

Comment:
I'm unsure why `caddy respond` needs this here to get a valid default config now, instead of line 209. `caddy run` works without it.